controller/controller_test.go

```diff
@@ -245,22 +248,23 @@ var _ = Describe("Controller", func() {
 				Expect(err).ToNot(HaveOccurred())
 				Expect(instanceIP).ToNot(BeNil())
 				Expect(instanceIP.GetUuid()).To(Equal(testInstanceIP.GetUuid()))
+				Expect(instanceIP.GetInstanceIpAddress()).ToNot(Equal(""))
 
-				// TODO: check if got an IP address (instance_ip_address)
-
+				existingIP, err := types.InstanceIpByUuid(client.ApiClient, instanceIP.GetUuid())
+				Expect(err).ToNot(HaveOccurred())
+				Expect(existingIP.GetUuid()).To(Equal(instanceIP.GetUuid()))
 			})
 		})
 		Context("when instance IP doesn't exist in Contrail", func() {
 			It("creates new instance IP", func() {
 				instanceIP, err := client.GetOrCreateInstanceIp(testNetwork, testInterface)
 				Expect(err).ToNot(HaveOccurred())
 				Expect(instanceIP).ToNot(BeNil())
+				Expect(instanceIP.GetInstanceIpAddress()).ToNot(Equal(""))
 
 				existingIP, err := types.InstanceIpByUuid(client.ApiClient, instanceIP.GetUuid())
 				Expect(err).ToNot(HaveOccurred())
 				Expect(existingIP.GetUuid()).To(Equal(instanceIP.GetUuid()))
```

Comment:
Why did you not check ip here? Checking Uuid is enough? (if yes, why are you checking not empty string in 263 line)